### PR TITLE
#9751 - RNA Builder: Missing "Switch to right/left" tooltips when hovering over the phosphate position picker options

### LINKED ISSUE #9751: RNA Builder: Missing "Switch to right/left" tooltips when hovering over the phosphate position picker options
Found while testing #9120 
**Steps to Reproduce:**

1. Switch to Macro mode
2. Expand the RNA builder in the RNA tab of the Monomer library
3. Select a phosphate for the preset so that both options (3' and 5') are available; logic may set the picker to 3' by default.
4. Hover over the position picker option that is not currently selected (e.g., if 3' is set, hover over 5')

**Actual Results**
No tooltip ("Switch to right"/"Switch to left") appears when hovering over the alternative phosphate position option in the picker. Also, there are no tooltips for the currently selected option: "Phosphate on the left" or "Phosphate on the right".

https://github.com/user-attachments/assets/bfb1467b-0ba2-44ae-8766-64322f24b453

**Expected Result:**
When one option is selected in the phosphate position picker, and the user hovers over the other option, a tooltip should appear indicating the action, either "Switch to right" or "Switch to left". Moreover, when the user hovers over the currently selected option, the tooltip "Phosphate on the left" or "Phosphate on the right" appears.

Image

**Found in Environment:**
- Ketcher Version 3.15.0-rc.1 Build at 2026-04-02; 15:41:30
- Indigo Version 1.43.0-rc.1.0-gd74653443-wasm32-wasm-clang-19.0.0
- Chrome Version 146.0.7680.178 (Official Build) (64-bit)
- Win11

### PR DESCRIPTION
**Pull request by mariam-khutuashvili**
## How the feature works? / How did you fix the issue?
(Screenshots, videos, or GIFs, if applicable)
fixed
## Summary
  The RNA Builder phosphate position picker (the two side-by-side icons inside the Phosphate slot) had no tooltips when hovering over its options unless they were disabled. Now hovering shows:

  - The currently selected option → **"Phosphate on the left"** / **"Phosphate on the right"**
  - The other (unselected) option → **"Switch to left"** / **"Switch to right"**

  The disabled-state tooltip ("Sugar must have R1, …" etc.) is preserved.

  ## Changes
  `packages/ketcher macromolecules/src/components/monomerLibrary/RnaBuilder/RnaEditor/RnaEditorExpanded/RnaEditorExpanded.tsx` — `renderPhosphatePositionOption` now picks the tooltip from one of three branches:

  - `isDisabled` → existing disabled message
  - enabled and matches `selectedPhosphatePosition` → reuses `getPhosphatePositionTooltip(position)` ("Phosphate on the left/right")
  - enabled and doesn't match → ``Switch to ${position}``

  `getPhosphatePositionTooltip` was moved above its first use site since `renderPhosphatePositionOption` now calls it.

  ## Test plan
  - [x] `prettier --check` clean on the changed file
  - [x] `eslint --quiet` clean
  - [x] `tsc --noEmit` clean for `ketcher-macromolecules` (only pre-existing `Cannot find module 'ketcher-react'` errors that exist on master)
  - [x] All 12 RnaBuilder unit tests pass; 3 snapshots unchanged
  - [x] Manual verification (Macromolecules → RNA tab → RNA Builder, build a preset like `R(A)P`):
    - [x] Hovering the **selected** phosphate-position option → "Phosphate on the right" tooltip
    - [x] Hovering the **other** option → "Switch to left" tooltip
    - [x] Switching positions and re-hovering produces the symmetric tooltips
    - [x] Disabled option still shows the existing "Sugar must have R1…" tooltip
    - [x] Master shows no tooltip on either picker option (confirmed bug)

## Check list
- [ ] unit-tests written
- [ ] e2e-tests written
- [ ] documentation updated
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name doesn't contain '#'
- [x] PR is linked with the issue
- [ ] base branch (master or release/xx) is correct
- [x] task status changed to "Code review"
- [ ] reviewers are notified about the pull request

### CODE REVIEW

File: packages/ketcher-macromolecules/src/components/monomerLibrary/RnaBuilder/RnaEditor/RnaEditorExpanded/RnaEditorExpanded.tsx [hunk 1]
```diff
@@ -419,34 +419,42 @@ export const RnaEditorExpanded = ({
...
-  );
+  ) => {
+    let tooltip: string;
+    if (isDisabled) {
+      tooltip = phosphatePositionDisabledTooltip[position];
+    } else if (selectedPhosphatePosition === position) {
+      tooltip = getPhosphatePositionTooltip(position);
+    } else {
+      tooltip = `Switch to ${position}`;
+    }
```
[the remark this answers is not in the document]
Comment (mariam-khutuashvili) [reply packages/ketcher-macromolecules/src/components/monomerLibrary/RnaBuilder/RnaEditor/RnaEditorExpanded/RnaEditorExpanded.tsx:436]: Autotests in ketcher-autotests/tests/specs/Macromolecule-editor/RNA-Builder/rna-library.spec.ts exercise the picker hover path; adding a unit test for the 3-branch logic would duplicate that coverage with heavy Redux mock setup for a Severity: Low fix. Leaving as-is.

**Review by rrodionov91 — APPROVED** [review]
